### PR TITLE
Add signup functionality and add profile functionality

### LINKED ISSUE #4: a user can setup a profile pic during signup or after login
1. UI 
2. Camera intent

### PR DESCRIPTION
**Pull request by robo360**
- Added signup functionality 
- Added profile setup functionality 

Note: * user can add profile from files or from camera
* bugs fixed

### CODE REVIEW

File: app/src/main/java/com/example/game/activities/SignupActivity.java [hunk 1]
```diff
@@ -1,36 +1,15 @@
...
-import androidx.fragment.app.FragmentTransaction;
 
 import android.os.Bundle;
-import android.util.Log;
-import android.view.View;
-import android.widget.Button;
-import android.widget.EditText;
-import android.widget.TextView;
-import android.widget.Toast;
+
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/SignupActivity.java:8]: nit:redundant new line.

File: app/src/main/java/com/example/game/fragments/SignupFragment.java [hunk 2]
```diff
@@ -38,7 +36,10 @@
     private EditText etRepeatedPassword;
     private Button btnSignup;
     private TextView tvLoginMessage;
-    private SignupFragment fragment;
+
+    public SignupFragment(){
+        // Required empty public constructor
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/SignupFragment.java]: could remove

File: app/src/main/java/com/example/game/fragments/TakePictureFragment.java [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    private ImageButton ibFile;
+    private ImageButton ibCamera;
+    private Button btnSubmit;
+    private ImageView ivProfile;
+    private TextView tvSkip;
+    private Bitmap image;
+    private File photoFile;
+
+    public TakePictureFragment() {
+        // Required empty public constructor
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/TakePictureFragment.java]: could remove

File: app/src/main/java/com/example/game/fragments/TakePictureFragment.java [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+                    save(ParseUser.getCurrentUser(), new ParseFile(photoFile));
+                }
+            }
+        });
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/TakePictureFragment.java:118]: could remove

File: app/src/main/java/com/example/game/fragments/TakePictureFragment.java [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        Intent intent = new Intent(Intent.ACTION_PICK,
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+        if (intent.resolveActivity(getContext().getPackageManager()) != null) {
+            startActivityForResult(intent, PICK_PHOTO_CODE);
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if ((data != null) && requestCode == PICK_PHOTO_CODE) {
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/TakePictureFragment.java]: could remove ()
```suggestion
        if (data != null && requestCode == PICK_PHOTO_CODE) {
```

File: app/src/main/java/com/example/game/fragments/TakePictureFragment.java [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+            Uri photoUri = data.getData();
+            // Load the image located at photoUri into selectedImage
+            image = loadFromUri(photoUri);
+            photoFile = new File(saveToInternalStorage(image));
+            ivProfile.setImageBitmap(image);
+        }
+        if (requestCode == CAPTURE_IMAGE_ACTIVITY_REQUEST_CODE) {
+            if (resultCode == RESULT_OK) {
+                // by this point we have the camera photo on disk
+                //Bitmap takenImage = BitmapFactory.decodeFile(photoFile.getAbsolutePath());
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/TakePictureFragment.java]: nit:cleanup debug code?

File: app/src/main/java/com/example/game/fragments/TakePictureFragment.java [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+                image = MediaStore.Images.Media.getBitmap(getContext().getContentResolver(), photoUri);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return image;
+    }
+
+    public File getPhotoFileUri(String fileName) {
+        // Get safe storage directory for photos
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/TakePictureFragment.java:172]: 👍

File: app/src/main/java/com/example/game/helpers/NavigationUtil.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
+package com.example.game.helpers;
+
+import android.app.Activity;
+import android.content.Intent;
+
+/*
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/helpers/NavigationUtil.java:6]: 👍 
If you want to use the javadoc style, use /** instead 😄

**Review by LynnHX — COMMENTED** [review]
Overall lg

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 9]
```diff
@@ -30,9 +30,9 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        if (ParseUser.getCurrentUser() != null) {
-            goToActivity(MainActivity.class);
-        }
+//        if (ParseUser.getCurrentUser() != null) {
+//            goToActivity(MainActivity.class);
+//        }
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java:35]: nit:clean up

File: app/src/main/java/com/example/game/activities/MainActivity.java [hunk 10]
```diff
@@ -28,9 +28,9 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         //Run some tests
-        createCommunity("Brazil vs Spain");
-        getCommunities();
-        getEvents();
+        //createCommunity("Brazil vs Spain");
+        //getCommunities();
+        //getEvents();
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/MainActivity.java:33]: nit:clean up

File: app/src/main/res/layout/fragment_signup.xml [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+    tools:context=".activities.LoginActivity">
+
+    <EditText
+        android:id="@+id/etName"
+        android:layout_width="0dp"
+        android:layout_height="@dimen/et_size"
+        android:layout_marginStart="@dimen/side_margin"
+        android:layout_marginTop="@dimen/top_margin"
+        android:layout_marginEnd="@dimen/side_margin"
+        android:autofillHints=""
```
Comment (LynnHX) [inline app/src/main/res/layout/fragment_signup.xml]: could remove

File: app/src/main/res/layout/fragment_signup.xml [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".activities.LoginActivity">
+
+    <EditText
+        android:id="@+id/etName"
+        android:layout_width="0dp"
+        android:layout_height="@dimen/et_size"
+        android:layout_marginStart="@dimen/side_margin"
+        android:layout_marginTop="@dimen/top_margin"
```
Comment (LynnHX) [inline app/src/main/res/layout/fragment_signup.xml:14]: 👍

File: app/src/main/res/layout/fragment_signup.xml [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+        android:layout_marginEnd="@dimen/side_margin"
+        android:autofillHints=""
+        android:background="@drawable/edit_text_border"
+        android:gravity="center"
+        android:hint="@string/name"
+        android:inputType="textMultiLine"
+        android:textAlignment="center"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.0"
+        android:textColorHint="@color/gray"
```
Comment (LynnHX) [inline app/src/main/res/layout/fragment_signup.xml:23]: could remove since we do not set hint

Comment (robo360) [reply app/src/main/res/layout/fragment_signup.xml:23]: We do have a hint

File: app/src/main/java/com/example/game/activities/LoginActivity.java (hunk 9 above)
Comment (robo360) [reply app/src/main/java/com/example/game/activities/LoginActivity.java:35]: resolved in the next pull request

**Review by LynnHX — APPROVED** [review]

**Pull request merged**